browser/brave_content_browser_client.cc

```diff
@@ -347,3 +334,54 @@ GURL BraveContentBrowserClient::GetEffectiveURL(
   return url;
 #endif
 }
+
+// [static]
+bool BraveContentBrowserClient::HandleURLOverrideRewrite(GURL* url,
+    content::BrowserContext* browser_context) {
+  // redirect sync-internals
+  if (url->host() == chrome::kChromeUISyncInternalsHost ||
+      url->host() == chrome::kChromeUISyncHost) {
+    GURL::Replacements replacements;
+    replacements.SetHostStr(chrome::kChromeUISyncHost);
+    *url = url->ReplaceComponents(replacements);
+    return true;
+  }
+
+  // no special win10 welcome page
+  if (url->host() == chrome::kChromeUIWelcomeWin10Host ||
+      url->host() == chrome::kChromeUIWelcomeHost) {
+    *url = GURL(chrome::kChromeUIWelcomeURL);
+    return true;
+  }
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+  if (url->SchemeIs(content::kChromeUIScheme) &&
+      url->host() == ethereum_remote_client_host) {
+    if (browser_context) {
+      extensions::ExtensionService* service =
```

Comment:
shouldn't this be removed since it was moved here? https://github.com/brave/brave-core/pull/2833/files#diff-fac884897c2bb076a63783b097c95c55R50

Reply:
@bridiver no because we don't want to rewrite the URL to the chrome-extension until after the throttle intercepts. The throttle won't hit the chrome-extension URL for an extension that is not installed.

Reply:
I'll change to ExtensionRegistry though and add a unit test to cover this aspect.

Reply:
OK done.

Reply:
did you push? Still shows the loader instead of ExtensionRegistry?

Reply:
I think we found you had a direct commit link which there was a change to.  Resolving this comment.